pkg/azuremachine/failure_domain.go

```diff
@@ -10,7 +10,7 @@ import (
 
 func validateFailureDomain(azureMachine capzv1alpha3.AzureMachine, supportedAZs []string) error {
 	// No failure domain specified.
-	if azureMachine.Spec.FailureDomain == nil {
+	if azureMachine.Spec.FailureDomain == nil || *azureMachine.Spec.FailureDomain == "" {
```

Comment:
I was thinking if it would help our future selves to use a constant called `AutomaticFailureDomainSelection` or something like that for the empty value.

Reply:
but what about nil?
It would be needed to have 2 constants, because nil and "" mean the same thing in this scenario

Reply:
Ok, never mind.